lib/backend/postgres/index.ts

```diff
@@ -739,15 +775,37 @@ export class PostgresBackend implements Database {
 						capabilities: [],
 					},
 					options,
-				),
-				this.upsertObject(
+				);
+			}
+
+			if (userContract.data.oauth && userContract.data.oauth !== {}) {
+				const authenticationContract = await this.upsertObject(
 					context,
 					{
-						slug: 'link-' + userContract.id + '-' + personalSettingsContract.id,
+						slug: 'authentication-oauth-' + userContract.slug,
+						version: userContract.version,
+						type: 'authentication@1.0.0',
+						active: userContract.active,
+						created_at: userContract.created_at,
+						markers: userContract.markers,
+						tags: [],
+						data: {
+							actorId: userContract.id,
+							oauth: userContract.data.oauth,
+						},
+						requires: [],
+						capabilities: [],
+					},
+					options,
+				);
+				await this.upsertObject(
+					context,
+					{
+						slug: 'link-' + userContract.id + '-' + authenticationContract.id,
 						version: '1.0.0',
 						type: 'link@1.0.0',
-						name: 'has attachment',
-						active: true,
+						name: 'is authenticated with',
```

Comment:
Is this a new link name? If so, we'll need to keep it in mind for the relationship work @ramirogm is tackling.